timemachine/cpp/src/wrap_kernels.cpp

```diff
@@ -842,6 +842,151 @@ void declare_potential(py::module &m) {
             du_dp has shape (coords_batch_size, param_batch_size, P)
             u has shape (coords_batch_size, param_batch_size)
 
+    )pbdoc")
+        .def(
+            "execute_batch_sparse",
+            [](Potential &pot,
+               const py::array_t<double, py::array::c_style> &coords,
+               const py::array_t<double, py::array::c_style> &params,
+               const py::array_t<double, py::array::c_style> &boxes,
+               const py::array_t<unsigned int, py::array::c_style> &coords_batch_idxs,
```

Comment:
Should add a check that `coords_batch_idxs` and `params_batch_idxs` have a min/max within 0 -> len(params)/len(coords) to avoid referencing invalid memory

Reply:
Added check in https://github.com/proteneer/timemachine/pull/1308/commits/a2e194b205c22898c159f3edfff2a8c70cb0a019